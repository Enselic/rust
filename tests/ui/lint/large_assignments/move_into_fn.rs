// build-fail

#![feature(large_assignments)]
#![move_size_limit = "1000"]
#![deny(large_assignments)]
#![allow(unused)]

fn main() {
    one_arg([0; 9999]); //~ ERROR large_assignments
    many_args(
        [0; 9999], //~ ERROR large_assignments
        true,      // no error
        [0; 9999], //~ ERROR large_assignments
    );
}

fn one_arg(a: [u8; 9999]) {}

fn many_args(a: [u8; 9999], b: bool, c: [u8; 9999]) {}
