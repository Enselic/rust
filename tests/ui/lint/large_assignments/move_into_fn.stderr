error: moving 9999 bytes
  --> $DIR/move_into_fn.rs:9:13
   |
LL |     one_arg([0; 9999]);
   |             ^^^^^^^^^ value moved from here
   |
   = note: The current maximum size is 1000, but it can be customized with the move_size_limit attribute: `#![move_size_limit = "..."]`
note: the lint level is defined here
  --> $DIR/move_into_fn.rs:5:9
   |
LL | #![deny(large_assignments)]
   |         ^^^^^^^^^^^^^^^^^

error: moving 9999 bytes
  --> $DIR/move_into_fn.rs:11:9
   |
LL |         [0; 9999],
   |         ^^^^^^^^^ value moved from here
   |
   = note: The current maximum size is 1000, but it can be customized with the move_size_limit attribute: `#![move_size_limit = "..."]`

error: moving 9999 bytes
  --> $DIR/move_into_fn.rs:13:9
   |
LL |         [0; 9999],
   |         ^^^^^^^^^ value moved from here
   |
   = note: The current maximum size is 1000, but it can be customized with the move_size_limit attribute: `#![move_size_limit = "..."]`

error: aborting due to 3 previous errors

